crates/rspack_core/src/cache/persistent/mod.rs

```diff
@@ -92,13 +105,17 @@ impl Cache for PersistentCache {
       .snapshot
       .add(modified_paths.iter().map(|item| item.as_ref()));
 
-    self.storage.trigger_save();
+    // TODO listen for storage finish in build mode
+    let _ = self.storage.trigger_save();
   }
 
-  fn before_make(&self, make_artifact: &mut MakeArtifact) {
+  async fn before_make(&self, make_artifact: &mut MakeArtifact) {
     if !make_artifact.initialized {
-      if let Ok(artifact) = self.make_occasion.recovery() {
-        *make_artifact = artifact;
+      match self.make_occasion.recovery().await {
+        Ok(artifact) => *make_artifact = artifact,
+        Err(err) => {
+          tracing::warn!("recovery error with {err:?}")
```

Comment:
can the error added into `compilation.diagnostics`?

Reply:
Of course, I will process the error and info in another PR. This PR aims to make persistent caching works.